flowey/flowey_lib_common/src/install_dist_pkg.rs

```diff
@@ -21,6 +26,98 @@ flowey_request! {
     }
 }
 
+#[derive(Debug)]
+struct PackageManager {
+    distro: FlowPlatformLinuxDistro,
+    sh: xshell::Shell,
+}
+
+impl PackageManager {
+    fn new(ctx: &NodeCtx<'_>) -> anyhow::Result<Self> {
+        let distro = match ctx.platform() {
+            FlowPlatform::Linux(linux_distribution) => linux_distribution,
+            _ => anyhow::bail!("Unsupported platform"),
+        };
+
+        let sh = xshell::Shell::new()?;
+
+        Ok(Self { distro, sh })
+    }
+
+    fn distro(&self) -> FlowPlatformLinuxDistro {
+        self.distro
+    }
+
+    // TODO: query_cmd should return a BTreeSet, ensuring that the processing of
```

Comment:
this isn't a TODO anymore though, right? this function _does_ return a BTreeSet now?

Reply:
Double-confirmed, fixed